jax/_src/lax/control_flow/common.py

```diff
@@ -58,17 +57,24 @@ def _initial_style_open_jaxpr(fun: Callable, in_tree, in_avals,
   wrapped_fun, out_tree = flatten_fun_nokwargs(lu.wrap_init(fun), in_tree)
   debug = pe.debug_info(fun, in_tree, out_tree, False,
                         primitive_name or "<unknown>")
-  jaxpr, _, consts, () = pe.trace_to_jaxpr_dynamic(wrapped_fun, in_avals, debug)
-  return jaxpr, consts, out_tree()
+  jaxpr, _, consts, attrs_tracked = pe.trace_to_jaxpr_dynamic(
+      wrapped_fun, in_avals, debug)
+  return jaxpr, consts, out_tree(), attrs_tracked
 
-@weakref_lru_cache
 def _initial_style_jaxpr(fun: Callable, in_tree, in_avals,
                          primitive_name: str | None = None):
-  jaxpr, consts, out_tree = _initial_style_open_jaxpr(
+  jaxpr, consts, out_tree, () = _initial_style_open_jaxpr(
       fun, in_tree, in_avals, primitive_name)
-  closed_jaxpr = core.ClosedJaxpr(pe.convert_constvars_jaxpr(jaxpr), ())
+  closed_jaxpr = pe.close_jaxpr(pe.convert_constvars_jaxpr(jaxpr))
   return closed_jaxpr, consts, out_tree
 
+def _initial_style_jaxpr_attrs(fun: Callable, in_tree, in_avals,
```

Comment:
This function is pretty redundant with the one above; the only difference is that it returns attrs. Instead of updating all callers of _initial_style_jaxpr, I decided just to introduce this alternative helper. Once we transition all callers to support attrs, we can delete the old one.